src/Agent/NewRelic/Agent/Core/Agent.cs

```diff
@@ -145,7 +150,21 @@ public ITransaction CreateTransaction(bool isWeb, string category, string transa
                 ? TransactionName.ForWebTransaction(category, transactionDisplayName)
                 : TransactionName.ForOtherTransaction(category, transactionDisplayName);
 
-            return CreateTransaction(initialTransactionName, doNotTrackAsUnitOfWork, wrapperOnCreate ?? NoOpWrapperOnCreate);
+            var transaction = CreateTransaction(initialTransactionName, doNotTrackAsUnitOfWork, wrapperOnCreate ?? NoOpWrapperOnCreate);
+
+            if (transactionNamePriority != (TransactionNamePriority)9999)
+            {
+                if (initialTransactionName.IsWeb)
+                {
+                    transaction.SetWebTransactionName(initialTransactionName.Category, initialTransactionName.Name, transactionNamePriority);
+                }
+                else
+                {
+                    transaction.SetOtherTransactionName(initialTransactionName.Category, initialTransactionName.Name, transactionNamePriority);
+                }
+            }
```

Comment:
I'm wondering if this logic is better off being handled within the service bus instrumentation directly. Right now that's the only scenario for it, and we might not have other scenarios.